haystack/preprocessor/utils.py

```diff
@@ -265,6 +266,12 @@ def fetch_archive_from_http(url: str, output_dir: str, proxies: Optional[dict] =
             elif url[-7:] == ".tar.gz":
                 tar_archive = tarfile.open(temp_file.name)
                 tar_archive.extractall(output_dir)
+            elif url[-3:] == ".gz":
```

Comment:
One small suggestion to use file magic number to determine file type. Sometime file extension changed by wget/curl. In this way we can have more generic way. Also able support other file format like what if file is not compressed?
In python https://github.com/ahupp/python-magic can be used which supported on Linux, and MacOS directly and Windows with extra lib support.

More information about file magic number https://en.wikipedia.org/wiki/List_of_file_signatures

Reply:
I think adding this change in this PR will increase PR size. So better to keep the change you did as it is. And improve this piece of code via file magic in separate PR along with good tests. WDYT?

Reply:
Hey @lalitpagaria thanks for the suggestion! It's a good one. I've done some small experiments on it. Will open a PR and share my findings